packages/transformer/src/IModelTransformer.ts

```diff
@@ -3225,13 +3238,50 @@ export class IModelTransformer extends IModelExportHandler {
       this._csFileProps.length === 0 ? "no-changes" : "has-changes";
   }
 
+  /**
+   * The behavior of process is influenced by the value of [[IModelTransformOptions.isSynchronization]] passed to the constructor of the IModelTransformer.
+   * @section When isSynchronization is true:
+   *
+   * Export changes from the source iModel and import the transformed entities into the target iModel.
+   * Inserts, updates, and deletes are determined by inspecting the changeset(s).
+   *
+   * Notes:
+   * - the transformer assumes that you saveChanges after processing changes. You should not modify the iModel after processChanges until saveChanges,
+   * failure to do so may result in corrupted
+   * data loss in future branch operations
+   * - if no startChangesetId or startChangeset option is provided as part of the ProcessChangesOptions, the next unsynchronized changeset
+   * will automatically be determined and used
+   * - To form a range of versions to process, set `startChangesetId` for the start (inclusive) of the desired range and open the source iModel as of the end (inclusive) of the desired range.
+   *
+   * @section When isSynchronization is false:
+   *
+   * Export everything from the source iModel and import the transformed entities into the target iModel.
+   *
+   * Notes:
+   * - [[processSchemas]] is not called automatically since the target iModel may want a different collection of schemas.
+   *
+   * @throws if [[IModelTransformOptions.isSynchronization]] is true and [[ProcessChangesOptions]] is not provided.
+   */
+  public async process(options?: ProcessChangesOptions): Promise<void> {
+    if (!this._initialized) {
```

Comment:
same `this._initialized` is validated in initialize(). Seems that we can skip it here

Reply:
Well actually initialize returns early if initialized is true, but this will call initialize for you if its currently false.

Reply:
My thinking is that `initialize ()` can be safely called as many times as you want, as it will initialize only once any way. No need to double check. Unless you have performance concerns to call `initialize ()`. Not sure if there is unit test for this behavior, but you could add one, to make sure it stays that way in future implementation:
`initialize () => should initialize only once when called multiple times`

Reply:
Oh I see what you're saying now, no if statement in process for the initialized var because initialize will return early anyways.

Reply:
removed the check in process